src/Application.vala

```diff
@@ -98,19 +108,42 @@ public class ValaLint.Application : GLib.Application {
                     continue;
                 }
 
-                Gee.ArrayList<FormatMistake?> mistakes = linter.run_checks_for_file (file);
-
-                if (!mistakes.is_empty) {
-                    command_line.print ("\x001b[1m\x001b[4m" + "%s" + "\x001b[0m\n", path);
+                check_file (file);
+            }
+        }
+    }
 
-                    foreach (FormatMistake mistake in mistakes) {
-                        command_line.print ("\x001b[0m%5i:%-3i \x001b[1m%-40s   \x001b[0m%s\n",
-                            mistake.line_index,
-                            mistake.char_index,
-                            mistake.mistake,
-                            mistake.check.get_title ());
-                    }
+    void check_directory (File dir) throws Error, IOError {
+        FileEnumerator enumerator = dir.enumerate_children (FileAttribute.STANDARD_NAME, 0, null);
+        var info = enumerator.next_file (null);
+        while (info != null) {
+            File child_file = dir.resolve_relative_path (info.get_name ());
+            if (info.get_file_type () == FileType.DIRECTORY) {
+                if (!info.get_is_hidden ()) {
+                    check_directory (child_file);
                 }
+            } else if (info.get_file_type () == FileType.REGULAR){
+                /* Check only .vala files */
+                if (info.get_name ().length > 5 && info.get_name ().substring (-5) == ".vala") {
```

Comment:
better create a variable `child_name = info.get_name ()` and reuse `child_name` it's consuming less CPU cycles.
Also consider using `child_name.has_suffix (".vala")` instead of black magic 😉

Reply:
You're right, it should be a little bit cleaner now 👍